### DISCUSSION TITLE
There is no problem with the MAC packaged app. Share other Macs. An error is reported and the app cannot be opened

### DISCUSSION THREAD

**Discussion by gffguo** [pyinstaller/pyinstaller#6508] [Help]
environment  
macOs Monterey
python 3.8
pyinstaller 4.7

Packaging command
pyinstaller -y --clean -w -i xx.ico  run.py

**Comment by gffguo** [comment]
![image](https://user-images.githubusercontent.com/69896971/149316237-0e55ee11-0bcc-47f8-a87f-51e0fc27b4c6.png)
Picture content: you don't have enough permission to open the app. Please contact the administrator

After I enter the app directory and use chmod - R 777 *, the app will flash back. Is there a path problem with my app package

**Comment by rokm** [comment]
Does the target Mac allow running unsigned 3rd party software (assuming your bundle is not signed), and/or software not installed from the app store (as per system policy)? 

How did you transfer the app bundle to the other Mac? Did it end up having quarantine flag by any chance? You can check by running `xattr -r program.app/` on the target Mac...

**Reply by JacobBumgarner** [reply]
So my package won't open after distribution to another computer, and when I run the `xattr -r program.app`, everything gets a quarantine flag. 

Can I fix this issue by code signing the app? If so, is [this resource](https://gist.github.com/txoof/0636835d3cc65245c6288b2374799c43) correct in saying that I'll have to have an apple developer account to sign the app?

**Reply by bwoodsend** [reply]
Yes and yes. Welcome to the misery that is developing for macOS.

**Reply by JacobBumgarner** [reply]
God. Is this true for all Macs? For example the one I'm testing it on is 'owned' by my university, and as such has a bunch of strict rules locked in. 

If someone just bought a Mac from the store, would it open an unsigned app out of the box?

**Reply by bwoodsend** [reply]
It will flat out refuse to open an unsigned app out the box nor is there any setting to disable that rule. You can right-click and select open on an unsigned executable in Finder (as opposed to just clicking on it) then it gives the user the option to allow that program to run. Other than that, there is no leeway.

**Reply by JacobBumgarner** [reply]
Hmmm I think I understand - So if I code sign the app with a self-signed certificate, it will just give the 'unknown developer' warning? But if this warning is ignored, the app can still be opened?

**Reply by JacobBumgarner** [reply]
And I think I'll have to freeze the app without a code sign, run [this script from the wiki](https://github.com/pyinstaller/pyinstaller/wiki/Recipe-OSX-Code-Signing-Qt), and then code sign from the terminal?

Sorry that I've hijacked this thread....

**Reply by bwoodsend** [reply]
Self signed certificates used to work but now are seen as even worse than not singed at all. There's no way around it - you have to pay.

**Reply by JacobBumgarner** [reply]
Gotcha - ok well then I guess I'll drop the $99.. I see why apple is such a cash cow now.

I tried adding a self-signed certificate arg to the pyinstaller build, but it failed because of the Qt issue. So I built without code signing and used the wiki script above. Then when I went to sign it from the console, it tells me that the app has already been signed. Thoughts/suggestions? Is pyinstaller signing it on its own?

And thanks for your help/responsiveness.

**Reply by bwoodsend** [reply]
There's a more up to date gist [here](https://gist.github.com/txoof/0636835d3cc65245c6288b2374799c43).

**Reply by rokm** [reply]
>  it tells me that the app has already been signed. Thoughts/suggestions? Is pyinstaller signing it on its own?

Yes, we perform ad-hoc signing by default (because of requirements on M1). But you can always force re-sign by adding `--force` to your `codesign` command...

**Reply by gffguo** [reply]
Hmmm, thank you, I think I should give up directly giving the package to others. 
Perhaps due to system reasons, there is no option open on an unsigned executable in finder in my finder right-click menu, Mac OS development is too painful.....

**Reply by JacobBumgarner** [reply]
@bwoodsend and @rokm ... Thanks for all of your patience and suggestions to help me to get my app working. This is clearly not my area of expertise, so I appreciate your input.

I'll test the distribution again today with your suggestions and will hopefully not have to ask for more help!

Cheers
